terraform_compliance/steps/steps.py

```diff
@@ -201,15 +202,21 @@ def its_key_is_value(_step_obj, key, value):
         elif isinstance(object_key, list) and value in object_key:
             found_list.append(obj)
 
-        elif isinstance(object_key, dict) and (value in object_key.keys()):
-            found_list.append(obj)
+        elif isinstance(object_key, dict):
+            if value in object_key.keys():
+                if dict_value is None or (object_key[value] == dict_value):
```

Comment:
Lets use `.lower()` on both `object_key[value]` and `dict_value` to avoid case sensitive problems. Of course then you have problems like what will it happen if the value is `bool` or `int`. Thus somewhere above, you need to check this and convert everything into `str` for ease the pain.

I want to move all these step logics in a better-engineered way to their own classes or functions. This is not specifically about this PR, mostly about me. I am sure you had hard times to understand what is going on on these steps :)

Reply:
I've lashed in some code to get some test running. Up for major refactor of both positive and negative methods.